### ISSUE TITLE
Bug: Pianobook Autumn Piano / monolithic libraries not loading

### ISSUE THREAD

**Issue by IsaakCode** [sfztools/sfizz#912]
Attempting to open the library results in indefinite loading. The plugin doesn't stop to load, so I have to shut down the DAW.

This is the library: https://www.pianobook.co.uk/library/autumn-piano/

I believe that specifically those libraries made by Christian Henson use a less common method: monolithic audio file with no separate samples per note, but one file containing all the notes with specific timestamps being referenced in the mapping. There seems to be a similar issue when loading the Decent Sampler version on that platform: 
https://www.pianobook.co.uk/forums/topic/having-trouble-with-christian-hensons-samples-in-decent-sampler-format/

VST 3 version on Windows, Ableton 11

**Comment by paulfd** [comment]
Yes as one can imagine, this method does not bode very well with the idea of "preloading" the start of the samples, since basically you have to preload the whole file. However, it should  still work, so I'll check it out. Thanks for the report.

**Comment by praashie** [comment]
Maybe monolith samples could be preloaded in individual chunks based on where each playable region actually starts?

**Comment by paulfd** [comment]
I made a change which should speed up this case up to bearable speed. This instrument seems to behave weirdly though so I'll check it a bit more.

> Maybe monolith samples could be preloaded in individual chunks based on where each playable region actually starts?

Yes that'd be the idea, although I have to admit that I was reluctant to slap this complexity on the file loading logic. Philosophically, I think we invented operating systems to handle storing chunks of data and referring to them with names and properties, and called them files. I tracked #936 to possibly improve this down the line.
